File: gaia-sdk-python/gaia_sdk/graphql/request/input/DeleteEdgeImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class DeleteEdgeImpulse():
    """
    The specification to delete an edge instance
    """
    source: str
    edgeId: str

    def __init__(self, source: str, edgeId: str):
        self.source = source
        self.edgeId = edgeId

    def __eq__(self, other):
        if type(other) is type(self):
            return self.source == other.source and self.edgeId == other.edgeId
        return False

    def __repr__(self):
        return {'source': self.source, 'edgeId': self.edgeId}

File: gaia-sdk-javascript/src/graphql/request/input/DeleteEdgeImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to delete an edge instance
 */
export class DeleteEdgeImpulse {
public _typeName = "DeleteEdgeImpulse";
    private source:Uuid;
    private edgeId:Uuid;

    constructor (source:Uuid, edgeId:Uuid) {
        this.source = source;
        this.edgeId = edgeId;
    }
}
